src/Controllers/MagicLinkController.php

```diff
@@ -77,14 +77,18 @@ public function loginAction()
 
         // Send the user an email with the code
         helper('email');
-        $email = emailer();
-        $email->setFrom(setting('Email.fromEmail'), setting('Email.fromName') ?? '')
+        $email     = emailer();
+        $emailSend = $email->setFrom(setting('Email.fromEmail'), setting('Email.fromName') ?? '')
             ->setTo($user->email)
             ->setSubject(lang('Auth.magicLinkSubject'))
             ->setMessage(view(setting('Auth.views')['magic-link-email'], ['token' => $token]))
             ->send();
 
-        return $this->displayMessage();
+        if ($emailSend) {
+            return $this->displayMessage();
+        }
+
+        return redirect()->route('magic-link')->with('error', lang('Auth.unableSendEmailToUser', [$user->email]));
```

Comment:
This code is different from Email2FA/EmailActivator:
https://github.com/codeigniter4/shield/blob/c66012f887d9e9ab14183472b162114f48ad4561/src/Authentication/Actions/Email2FA.php#L80-L82

The error message is user friendly, but it seems there is no log of the mail sending failure.

Reply:
@kenjis , I really disagree with that.
I like the error message is user friendly.
Why? Just imagine yourself instead of the user.
If necessary, the site administrator should see the logs of file **writable\logs**.
Friends, **what do you think about this?**
**Do you agree with the following code?**
```
 if ($return === false) { 
     throw new RuntimeException('Cannot send email for user: ' . $user->email); 
 } 
```

Reply:
I forgot that the Email class logs properly.

Okay, this implementation has no problem.

Reply:
Can you change the code like this (the variable name) as with other classes?
```php
$return = emailer()->setFrom(setting('Email.fromEmail'), setting('Email.fromName') ?? '')
```
https://github.com/codeigniter4/shield/blob/c66012f887d9e9ab14183472b162114f48ad4561/src/Authentication/Actions/Email2FA.php#L74

Reply:
**Done.**